components/application-operator/pkg/kymahelm/application/releasemanager.go

```diff
@@ -48,6 +56,37 @@ func (r *releaseManager) InstallChart(application *v1alpha1.Application) (hapi_4
 	return installResponse.Release.Info.Status.Code, installResponse.Release.Info.Description, nil
 }
 
+func (r *releaseManager) UpgradeReleases() error {
+	appList, err := r.appClient.List(v1.ListOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "Error fetching application list")
+	}
+
+	for _, app := range appList.Items {
+		status, description, err := r.upgradeChart(&app)
+		if err != nil {
+			setCurrentStatus(&app, status.String(), description)
+			r.appClient.Update(&app)
```

Comment:
I think it would be good to use retry on conflict mechanism and log some warning if we fail.